### PR TITLE
fix(zero): prevent multiple exports

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Changes

- Prevent multiple exports in entrypoints
This prevents people from exporting functions (and else) and reusing them in other syncs, causing issues with double exports.
This might be a breaking change for some people exporting types, but it avoids bigger issues.

---

**Restrict Multiple Named Exports in Zero Entrypoints**

This PR enhances the code generation validation for zero-based entrypoints by prohibiting multiple named exports, only permitting defaults or the named exports `createAction`, `createSync`, and `createOnEvent`. It introduces AST-level checks in the Babel plugin, a new error code, expanded unit tests and new fixture coverage, and performs a minor unrelated cleanup in a backend route. These changes are aimed at preventing issues caused by re-exporting functions (or other symbols) between syncs, which previously could lead to unpredictable or duplicated behavior.

<details>
<summary><strong>Key Changes</strong></summary>

• Added AST-level validation to the Babel plugin in `packages/cli/lib/zeroYaml/compile.ts` to allow only default exports and specific named exports (`createAction`, `createSync`, `createOnEvent`) in zero entrypoint files.
• Permits re-exports using `export { ... } from ...` but otherwise throws a new `nango_named_export_not_allowed` error for violations.
• Introduced `AugmentedExport` and `AugmentedExportDefault` TypeScript types for safer internal handling in the plugin.
• Added new test fixture `packages/cli/fixtures/zero/cases/multipleExports.ts` demonstrating a forbidden export pattern.
• Expanded unit tests in `packages/cli/lib/zeroYaml/compile.unit.cli-test.ts` to assert error generation for illegal exports.
• Registered error code `nango_named_export_not_allowed` in `packages/cli/lib/zeroYaml/utils.ts`.
• Removed an unused `console.log` statement from `packages/jobs/lib/routes/tasks/putTask.ts`.

</details>

<details>
<summary><strong>Affected Areas</strong></summary>

• `packages/cli/lib/zeroYaml/compile.ts` (Babel plugin, export logic)
• `packages/cli/fixtures/zero/cases/multipleExports.ts` (test fixture)
• `packages/cli/lib/zeroYaml/compile.unit.cli-test.ts` (unit tests)
• `packages/cli/lib/zeroYaml/utils.ts` (error enumeration)
• `packages/jobs/lib/routes/tasks/putTask.ts` (minor cleanup)

</details>

---
*This summary was automatically generated by @propel-code-bot*

### CODE REVIEW

File: packages/cli/lib/zeroYaml/compile.ts [hunk 1]
```diff
@@ -478,6 +478,71 @@ function nangoPlugin({ entryPoint }: { entryPoint: string }) {
...
+                                    const exportedName = specifier.exported.name;
+                                    if (!allowedExports.includes(exportedName)) {
+                                        throw new CompileError(
+                                            'nango_named_export_not_allowed',
+                                            lineNumber,
+                                            `Named export '${exportedName}' is not allowed. Only export default and ${allowedExports.join(', ')} are permitted.`
+                                        );
+                                    }
+                                }
+                            }
```
Comment (TBonnin) [inline packages/cli/lib/zeroYaml/compile.ts:524]: can we return here. if `node.specifier`, can it ever enter the if `node.declaratoin` below?

File: packages/cli/lib/zeroYaml/compile.ts [hunk 2]
```diff
@@ -478,6 +478,71 @@ function nangoPlugin({ entryPoint }: { entryPoint: string }) {
                         }
                     },
 
+                    ExportNamedDeclaration(astPath) {
+                        // Skip internally transformed exports
+                        if ((astPath.node as any).__transformedByRemoveCreateWrappers) {
```
Comment (TBonnin) [inline packages/cli/lib/zeroYaml/compile.ts]: no way to avoid the any here?

File: packages/cli/lib/zeroYaml/compile.ts [hunk 3]
```diff
@@ -478,6 +478,71 @@ function nangoPlugin({ entryPoint }: { entryPoint: string }) {
...
+                                    }
+                                }
+                            }
+
+                            for (const exportedName of exportedNames) {
+                                if (!allowedExports.includes(exportedName)) {
+                                    throw new CompileError(
+                                        'nango_named_export_not_allowed',
+                                        lineNumber,
+                                        `Named export '${exportedName}' is not allowed. Only export default and ${allowedExports.join(', ')} are permitted.`
```
Comment (TBonnin) [inline packages/cli/lib/zeroYaml/compile.ts]: nit: error is duplicated. Can we have a `namedExportError(...) => CompileError`

**Review by TBonnin — APPROVED** [review]

File: packages/cli/lib/zeroYaml/compile.unit.cli-test.ts [hunk 4]
```diff
@@ -57,4 +58,20 @@ describe('edge cases', () => {
...
+                new CompileError(
+                    'nango_named_export_not_allowed',
+                    25,
+                    "Named export 'test' is not allowed. Only export default and createAction, createSync, createOnEvent are permitted.",
+                    './zero/cases/multipleExports.ts'
+                )
+            );
+        } else {
+            throw new Error('should be an error');
+        }
```
Comment (kaposke) [inline packages/cli/lib/zeroYaml/compile.unit.cli-test.ts]: I think we can use assertions for these
```ts
assert(result.isErr(), 'Should be an error');
// Typescript understands it's an error from here
```

**Review by kaposke — APPROVED** [review]

File: packages/cli/lib/zeroYaml/compile.ts (hunk 1 above)
Comment (bodinsamuel) [reply packages/cli/lib/zeroYaml/compile.ts:524]: yes it can

File: packages/cli/lib/zeroYaml/compile.ts (hunk 2 above)
Comment (bodinsamuel) [reply packages/cli/lib/zeroYaml/compile.ts]: added a better interface

File: packages/cli/lib/zeroYaml/compile.unit.cli-test.ts (hunk 4 above)
Comment (bodinsamuel) [reply packages/cli/lib/zeroYaml/compile.unit.cli-test.ts]: ah nice it's type checking correctly 👍🏻
